stablehlo/tests/stablehlo_legalize_deprecated_ops.mlir

```diff
@@ -0,0 +1,76 @@
+// RUN: stablehlo-opt --stablehlo-legalize-deprecated-ops --split-input-file --verify-diagnostics %s | FileCheck %s
+
+// CHECK-LABEL: op_dot
+func.func @op_dot(%arg0: tensor<2x3xf32>,
+                 %arg1: tensor<3x?xf32>) -> tensor<2x?xf32> {
+  // CHECK: stablehlo.dot_general %arg0, %arg1, contracting_dims = [1] x [0] : (tensor<2x3xf32>, tensor<3x?xf32>) -> tensor<2x?xf32>
+  %0 = "stablehlo.dot"(%arg0, %arg1) : (tensor<2x3xf32>, tensor<3x?xf32>) -> tensor<2x?xf32>
+  func.return %0 : tensor<2x?xf32>
+}
+
+// CHECK-LABEL: op_unary_einsum
+func.func @op_unary_einsum(%arg0: tensor<8x16xf32>) -> tensor<8xf32> {
+  // CHECK: stablehlo.einsum %cst, %arg0, config = ",ab->a" : (tensor<f32>, tensor<8x16xf32>) -> tensor<8xf32>
```

Comment:
Should the constant be checked too? If not, consider using a regex to match the `cst` part.